core/time/src/clock.rs

```diff
@@ -134,9 +161,14 @@ impl FakeClockInner {
         }
         self.instant += d;
         self.utc += d;
+        
+        // Wake up any waiters that have reached their deadline
         while let Some(earliest_waiter) = self.waiters.peek() {
             if earliest_waiter.deadline <= self.instant {
-                self.waiters.pop().unwrap().waker.send(()).ok();
+                let waiter = self.waiters.pop().unwrap();
+                if waiter.waker.send(()).is_err() {
```

Comment:
I realise that the previous code was not failing if send failed.  I think that is not good and that we should take the same approach as receive below.  i.e. we should use `unwrap` here.